app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,14 @@
+class DishesController <ApplicationController
+  def show
+    @dish = Dish.find(params[:id])
+    @chef = Chef.find(params[:chef_id])
+  end
+
+  def update
+    @dish = Dish.find(params[:id])
+    @ingredient = Ingredient.find(params[:add_ingredient])
+    @dish.ingredients << @ingredient
+    redirect_to "/chef/#{@dish.chef_id}/dishes/#{@dish.id}"
+  end
```

Comment:
Let me start by just mentioning that since the `update` action doesn't send anything to a view, you don't need to set everything to an instance variable.

The only other thing I would call out is that this method, `update`, should be `create` and it should live on either the child controller `IngredientsController` or a join controller `DishIngredientsController`. If we think about what it's doing in the background it is creating a relationship between a child and the parent, it's not updating the parent.

The code itself is perfect. The redirect should be going to the dishes show page `redirect_to "/dishes/#{@dish.id}"`.